filters.go

```diff
@@ -157,3 +159,38 @@ func (p *Pipe) First(lines int) *Pipe {
 	}
 	return Echo(output.String())
 }
+
+// Freq reads from the pipe, and returns a new pipe containing only unique lines
+// from the input, prefixed with a frequency count, in descending numerical
+// order (most frequent lines first). Lines with equal frequency will be sorted
+// alphabetically. If there is an error reading the pipe, the pipe's error
+// status is also set.
+func (p *Pipe) Freq() *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	var freq = map[string]int{}
+	p.EachLine(func(line string, out *strings.Builder) {
+		freq[line]++
+	})
+	type frequency struct {
+		line  string
+		count int
+	}
+	var freqs = make([]frequency, 0, len(freq))
+	for line, count := range freq {
+		freqs = append(freqs, frequency{line, count})
+	}
+	sort.Slice(freqs, func(i, j int) bool {
+		if freqs[i].count == freqs[j].count {
+			return freqs[i].line < freqs[j].line
+		}
+		return freqs[i].count > freqs[j].count
+	})
+	var output strings.Builder
+	for _, item := range freqs {
```

Comment:
As the slice contains the line with the highest fequency as the first element, we can use the length of the frequency in digits to prefix all lines accordingly with something along these lines (given, we check the len already on line 176):

```
digits := len(strconv.Itoa(freqs[0]))
for _, item := range freqs {
        output.WriteString(fmt.Sprintf(fmt.Sprintf("%% %dd %%s", digits), item.count, item.line))
```
```

Reply:
That's roughly what I ended up doing on the other branch:

```go
	fieldWidth := len(strconv.Itoa(maxCount))
	...
		output.WriteString(fmt.Sprintf("%*d %s", fieldWidth, item.count, item.line))
```

(I was quite pleased to discover the `%*` construct—exactly what I needed!)

Reply:
Great catch, i did not know about `%*`.

Reply:
Neither did I—it's _almost_ undocumented, but it is there in the small print of `fmt`, if you use a large magnifying glass.